DuckDuckGo/AppDelegate.swift

```diff
@@ -903,22 +908,27 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return window?.rootViewController as? MainViewController
     }
 
-    func refreshShortcuts() {
+    @MainActor
+    func refreshShortcuts() async {
 #if NETWORK_PROTECTION
         guard vpnFeatureVisibility.shouldShowVPNShortcut() else {
             UIApplication.shared.shortcutItems = nil
             return
         }
 
-        let items = [
-            UIApplicationShortcutItem(type: ShortcutKey.openVPNSettings,
-                                      localizedTitle: UserText.netPOpenVPNQuickAction,
-                                      localizedSubtitle: nil,
-                                      icon: UIApplicationShortcutIcon(templateImageName: "VPN-16"),
-                                      userInfo: nil)
-        ]
+        if case .success = await AccountManager().hasEntitlement(for: .networkProtection) {
```

Comment:
Nit: Avoid hitting the backend in all cases

```suggestion
        if case .success = await AccountManager().hasEntitlement(for: .networkProtection, cachePolicy: .returnCacheDataDontLoad) {
```

Reply:
Done, and updated to make sure we check `success(true)` as well.